router/migrate_test.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"os"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/jackc/pgx"
+	"github.com/flynn/flynn/pkg/postgres"
+	"github.com/flynn/flynn/pkg/testutils/postgres"
+	"github.com/flynn/flynn/router/types"
+
+	. "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+)
+
+func setupTestDB(c *C, dbname string) *postgres.DB {
+	if err := pgtestutils.SetupPostgres(dbname); err != nil {
+		c.Fatal(err)
+	}
+	pgxpool, err := pgx.NewConnPool(pgx.ConnPoolConfig{
+		ConnConfig: pgx.ConnConfig{
+			Host:     os.Getenv("PGHOST"),
+			Database: dbname,
+		},
+	})
+	if err != nil {
+		c.Fatal(err)
+	}
+	return postgres.New(pgxpool, nil)
+}
+
+type MigrateSuite struct{}
+
+var _ = Suite(&MigrateSuite{})
+
+type testMigrator struct {
+	c  *C
+	db *postgres.DB
+	id int
+}
+
+func (t *testMigrator) migrateTo(id int) {
+	t.c.Assert((*migrations)[t.id:id].Migrate(t.db), IsNil)
+	t.id = id
+}
+
+func (MigrateSuite) TestMigrateTLSObject(c *C) {
+	db := setupTestDB(c, "routertest_tls_object_migration")
+	m := &testMigrator{c: c, db: db}
+
+	// start from ID 4
+	m.migrateTo(4)
+
+	cert := tlsConfigForDomain("migrationtest.example.com")
+	r := &router.Route{
+		ParentRef:     "some/parent/ref",
+		Service:       "migrationtest.example.com",
+		Domain:        "migrationtest.example.com",
+		LegacyTLSCert: cert.Cert,
+		LegacyTLSKey:  cert.PrivateKey,
+	}
+	err := db.Exec(`INSERT INTO http_routes (parent_ref, service, domain, tls_cert, tls_key) VALUES ($1, $2, $3, $4, $5)`, r.ParentRef, r.Service, r.Domain, r.LegacyTLSCert, r.LegacyTLSKey)
+	c.Assert(err, IsNil)
```

Comment:
Leading and trailing spaces and newlines are now stripped in both migration and main codepaths.